src/dhtproto/client/legacy/internal/helper/Handshake.d

```diff
@@ -215,27 +215,37 @@ unittest
             theScheduler.await(this.handshake.oneNodeConnected());
             Stdout.formatln("At least one node is now connected!");
 
-            // if `theScheduler.awaitOrTimeout` is available, it can
-            // be used to implement support for a partial handshake:
+            // wait until either all nodes have connected, or 60 seconds
+            // have passed, whichever comes sooner
             //
-            // ```
-            // auto timeout_microsec = 60_000_000;
-            //
-            // auto handshake_timed_out =   /* true if timeout is reached */
-            //     theScheduler.awaitOrTimeout(
-            //         this.handshake.allNodesConnected(),
-            //         timeout_microsec);
-            //
-            // /* react to `handshake_timed_out` as you wish, but note
-            //    that the `DhtHandshake` instance will keep working
-            //    in the background to complete the handshake */
-            // ```
-            //
-            // alternatively, we can block on all DHT nodes being
-            // connected using `await`:
-            theScheduler.await(this.handshake.allNodesConnected());
-
-            Stdout.formatln("DHT handshake complete!");
+            // (if you really need a hard block until the full handshake
```

Comment:
If you wanted that behaviour, there'd be no reason to call `oneNodeConnected` at all, right?

Reply:
Well, it depends.  In apps I'm responsible for, I'll probably make it a configurable option whether to `await` or `awaitOrTimeout` on all nodes being connected.  So the separation still has some meaning.

Reply:
Fair enough. The comment is innocent enough. I just wondered.